File: LeetCode/squares_of_a_sorted_array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        map<int, int> mp;
        
        for(int x : nums) {
            int a = pow(x, 2);
            mp[a]++;
        }
        
        vector<int> res;
        for(auto it : mp) {
            int size = it.second;
            while(size--) {
                res.push_back(it.first);
            }
        }
        
        return res;
    }
};

File: LeetCode/squares_of_a_sorted_array.go
func abs(a int) int {
    if a >= 0 {
        return a
    }

    return -a
}

func sortedSquares(nums []int) []int {
    ans := make([]int, len(nums))
    l, r := 0, len(nums)-1

    for i := len(nums)-1; i >= 0; i-- {
        if abs(nums[l]) > abs(nums[r]) {
            ans[i] = nums[l] * nums[l]
            l++
        } else {
            ans[i] = nums[r] * nums[r]
            r--
        }
    }

    return ans
}
